src/components/react-hook-form/numbers/float-input.tsx

```diff
@@ -48,14 +58,16 @@ const FloatInput = (props) => {
             // our acceptValue because they are needed as intermediate strings
             // for the user to input useful numbers.
             // TODO can we remove the isNaN check and the special cases check?
-            if (['-', '.'].includes(value)) {
-                return value;
+            if (['-', '.'].includes(value as string)) {
+                return value as string;
             }
-            return value === null || isNaN(value) ? '' : value;
+            return value === null || Number.isNaN(value)
+                ? ''
+                : (value as string);
```

Comment:
It changes the return type to `string | undefined` tho, have to add a `return '';` outside of the block for this to work

Reply:
Maybe add a small comment to explain why this condition ?

Reply:
Added the following 
>             //We explicitly test for 'string' type for code clarity, it enables
>             // use to avoid casting the value variable from 'number | string'
>             // to 'string' on multiple statements